server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServerCallbacks.java

```diff
@@ -54,17 +55,42 @@ default void onStreamOpen(long streamId, String typeUrl) {
    *     will be returned to the client and the stream will be closed with error.
    */
   default void onStreamRequest(long streamId, DiscoveryRequest request) {
+  }
 
+  /**
+   * {@code onV3StreamRequest} is called for each {@link io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest}
+   * that is received on the stream.
+   *
+   * @param streamId an ID for this stream that is only unique to this discovery server instance
+   * @param request the discovery request sent by the envoy instance
+   *
+   * @throws RequestException optionally can throw {@link RequestException} with custom status. That status
+   *     will be returned to the client and the stream will be closed with error.
+   */
+  default void onV3StreamRequest(long streamId, io.envoyproxy.envoy.service.discovery.v3.DiscoveryRequest request) {
```

Comment:
should I not default this for safety?

Reply:
Can you explain more what the consequences are?

Reply:
I believe could mean a user accidentally doesn't implement the new callback after doing an upgrade, and then they don't get the callbacks they expect.